packages/bpk-component-infinite-scroll/src/withInfiniteScroll.js

```diff
@@ -0,0 +1,239 @@
+ */
+
+/* @flow */
+
+import React, { Component, type Element, type ComponentType } from 'react';
+import PropTypes from 'prop-types';
+import { cssModules } from 'bpk-react-utils';
+import omit from 'lodash/omit';
+import 'intersection-observer';
+
+import STYLES from './withInfiniteScroll.scss';
+
+const getClassNames = cssModules(STYLES);
+
+type ScrollEvent = {
+  currentIndex: number,
+};
+
+type ScrollFinishedEvent = {
+  totalNumberElements: number,
+};
+
+type OnItemsFetchFunction = (
+  index: number,
+  elementsPerScroll: number,
+) => Promise<Array<any>>;
+
+export type Props = {
+  elementsPerScroll: number,
+  loadingComponent: ?Element<any>,
+  onItemsFetch: OnItemsFetchFunction,
+  onClickSeeMore: (event: SyntheticEvent<HTMLButtonElement>) => void,
+  onScroll: (o: ScrollEvent) => void,
+  onScrollFinished: (o: ScrollFinishedEvent) => void,
+  seeMoreComponent: Element<any>,
+  seeMoreAfter: number,
+};
+
+export type State = {
+  index: number,
+  elementsToRender: Array<any>,
+  isListFinished: boolean,
+  showSeeMore: boolean,
+};
+
+type extendedProps = {
+  elements: Array<any>,
+};
+
+const withInfiniteScroll = (
+  ComponentToExtend: ComponentType<extendedProps>,
+): ComponentType<Props> =>
+  class extends Component<Props, State> {
+    handleIntersection: IntersectionObserverCallback;
+    handleKeyPress: (e: SyntheticKeyboardEvent<HTMLButtonElement>) => void;
+    handleSeeMoreClick: (e: SyntheticEvent<HTMLButtonElement>) => void;
+    observer: IntersectionObserver;
+    onItemsFetch: OnItemsFetchFunction;
+    onScroll: (o: ScrollEvent) => void;
+    sentinel: ?HTMLElement;
+
+    static propTypes = {
+      elementsPerScroll: PropTypes.number,
+      loadingComponent: PropTypes.node,
+      onItemsFetch: PropTypes.func.isRequired,
+      onClickSeeMore: PropTypes.func,
+      onScroll: PropTypes.func,
+      onScrollFinished: PropTypes.func,
+      seeMoreComponent: PropTypes.node,
+      seeMoreAfter: PropTypes.number,
+    };
+
+    static defaultProps = {
+      elementsPerScroll: 5,
+      loadingComponent: null,
+      onClickSeeMore: () => {},
+      onScroll: () => {},
+      onScrollFinished: () => {},
+      seeMoreComponent: null,
+      seeMoreAfter: null,
+    };
+
+    constructor(props: Props) {
+      super(props);
+
+      this.state = {
+        index: 0,
+        elementsToRender: [],
+        isListFinished: false,
+        showSeeMore: false,
+      };
+
+      this.onItemsFetch = props.onItemsFetch;
+      this.onScroll = props.onScroll;
+      this.observer = new IntersectionObserver(this.handleIntersection, {
+        threshold: 1,
+      });
+    }
+
+    componentDidMount() {
+      if (this.sentinel) {
+        this.observer.observe(this.sentinel);
+      }
+    }
+
+    componentDidUpdate() {
+      if (this.sentinel && this.state.index > 0) {
+        this.observer.observe(this.sentinel);
+      }
+    }
+
+    loadMore() {
+      return new Promise(resolve => {
```

Comment:
You don't need to return a new promise here, just returning the promise bellow is enough

Reply:
done